### ISSUE TITLE
Build error after upgrading to Spring Boot 3.0.0-M5 in my app

### ISSUE THREAD

**Issue by wenqiglantz** [snicoll/demo-aot-native#1]
Hi Stéphane,

I am following your sample app `demo-aot-native` and tried to get my simple app to have native support out of the box with Spring Boot 3.0.0-M5, but I am running into native image build error, which I need your guidance to help resolve.  My app repo location is https://github.com/wenqiglantz/spring-native-feign-reactive.  The error I ran into when I ran "`mvn clean package -Pnative`" is as follows.  The cause seems to be `UnsupportedFeatureException: No instances of ch.qos.logback.classic.Logger are allowed in the image heap as this class should be initialized at image runtime`.  I didn't find anything in particular in your app related to logback hints configuration.  I would think Spring Boot 3.0.0-M5 should already have logback native hints built in.  Not sure why this error is thrown.  Any advice would be greatly appreciated!

```
Warning: Could not resolve condition io.netty.internal.tcnative.SSL for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSL.
Warning: Could not resolve condition io.netty.internal.tcnative.SSL for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSL.
Warning: Could not resolve condition io.netty.channel.udt.DefaultUdtChannelConfig for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.DefaultUdtChannelConfig.
Warning: Could not resolve com.github.luben.zstd.ZstdCompressCtx for reflection configuration. Reason: java.lang.ClassNotFoundException: com.github.luben.zstd.ZstdCompressCtx.
Warning: Could not resolve io.netty.handler.ssl.OpenSslClientSessionCache for reflection configuration. Reason: java.lang.NoClassDefFoundError: io/netty/internal/tcnative/SSLSessionCache.
Warning: Could not resolve condition io.netty.internal.tcnative.SSLContext for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSLContext.
Warning: Could not resolve io.netty.handler.ssl.ReferenceCountedOpenSslClientContext$ExtendedTrustManagerVerifyCallback for reflection configuration. Reason: java.lang.NoClassDefFoundError: io/netty/internal/tcnative/CertificateVerifier.
Warning: Could not resolve condition io.netty.internal.tcnative.SSLContext for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSLContext.
Warning: Could not resolve io.netty.handler.ssl.ReferenceCountedOpenSslServerContext$ExtendedTrustManagerVerifyCallback for reflection configuration. Reason: java.lang.NoClassDefFoundError: io/netty/internal/tcnative/CertificateVerifier.
Warning: Could not resolve io.netty.handler.ssl.ReferenceCountedOpenSslServerContext$OpenSslServerCertificateCallback for reflection configuration. Reason: java.lang.NoClassDefFoundError: io/netty/internal/tcnative/CertificateCallback.
Warning: Could not resolve condition io.netty.internal.tcnative.SSLContext for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSLContext.
Warning: Could not resolve condition io.netty.channel.udt.DefaultUdtChannelConfig for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.DefaultUdtChannelConfig.
Warning: Could not resolve condition io.netty.channel.udt.nio.NioUdtAcceptorChannel for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.nio.NioUdtAcceptorChannel.
Warning: Could not resolve condition io.netty.channel.udt.nio.NioUdtByteConnectorChannel for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.nio.NioUdtByteConnectorChannel.
Warning: Could not resolve condition io.netty.channel.udt.nio.NioUdtMessageConnectorChannel for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.nio.NioUdtMessageConnectorChannel.
Warning: Could not resolve condition io.netty.channel.udt.nio.NioUdtProvider for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.nio.NioUdtProvider.
Warning: Could not resolve condition io.netty.internal.tcnative.SSLContext for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.SSLContext.
Warning: Could not resolve condition io.netty.channel.udt.DefaultUdtChannelConfig for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.DefaultUdtChannelConfig.
Warning: Could not resolve org.conscrypt.ConscryptEngine for reflection configuration. Reason: java.lang.ClassNotFoundException: org.conscrypt.ConscryptEngine.
Warning: Could not resolve org.conscrypt.ConscryptEngine for reflection configuration. Reason: java.lang.ClassNotFoundException: org.conscrypt.ConscryptEngine.
Warning: Could not resolve org.conscrypt.ConscryptEngine for reflection configuration. Reason: java.lang.ClassNotFoundException: org.conscrypt.ConscryptEngine.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.handler.codec.stomp.DefaultStompHeaders for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.stomp.DefaultStompHeaders.
Warning: Could not resolve condition io.netty.handler.codec.stomp.DefaultStompHeaders for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.stomp.DefaultStompHeaders.
Warning: Method ch.qos.logback.core.encoder.LayoutWrappingEncoder.setParent(Appender) not found.
Warning: Could not resolve com.aayushatharva.brotli4j.Brotli4jLoader for reflection configuration. Reason: java.lang.ClassNotFoundException: com.aayushatharva.brotli4j.Brotli4jLoader.
Warning: Could not resolve com.github.luben.zstd.Zstd for reflection configuration. Reason: java.lang.ClassNotFoundException: com.github.luben.zstd.Zstd.
Warning: Could not resolve com.ning.compress.lzf.impl.UnsafeChunkDecoder for reflection configuration. Reason: java.lang.ClassNotFoundException: com.ning.compress.lzf.impl.UnsafeChunkDecoder.
Warning: Could not resolve condition io.netty.channel.sctp.DefaultSctpServerChannelConfig for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.sctp.DefaultSctpServerChannelConfig.
Warning: Could not resolve condition io.netty.channel.udt.DefaultUdtServerChannelConfig for reflection. Reason: java.lang.ClassNotFoundException: io.netty.channel.udt.DefaultUdtServerChannelConfig.
Warning: Could not resolve condition io.netty.handler.codec.redis.FixedRedisMessagePool for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.redis.FixedRedisMessagePool.
Warning: Could not resolve condition io.netty.handler.codec.smtp.SmtpRequestEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.smtp.SmtpRequestEncoder.
Warning: Could not resolve condition io.netty.handler.proxy.ProxyServer for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.proxy.ProxyServer.
Warning: Could not resolve condition io.netty.handler.proxy.ProxyServer$IntermediaryHandler for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.proxy.ProxyServer$IntermediaryHandler.
Warning: Could not register io.netty.handler.codec.compression.Lz4FrameDecoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: net/jpountz/lz4/LZ4Exception.
Warning: Could not register io.netty.handler.codec.compression.Lz4FrameEncoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: net/jpountz/lz4/LZ4Exception.
Warning: Could not resolve condition io.netty.handler.codec.haproxy.HAProxyMessageDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.haproxy.HAProxyMessageDecoder.
Warning: Could not resolve condition io.netty.handler.codec.haproxy.HAProxyMessageEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.haproxy.HAProxyMessageEncoder.
Warning: Could not register io.netty.handler.codec.marshalling.CompatibleMarshallingDecoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: org/jboss/marshalling/ByteInput.
Warning: Could not register io.netty.handler.codec.marshalling.CompatibleMarshallingEncoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: org/jboss/marshalling/ByteOutput.
Warning: Could not register io.netty.handler.codec.marshalling.MarshallingDecoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: org/jboss/marshalling/ByteInput.
Warning: Could not register io.netty.handler.codec.marshalling.MarshallingEncoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: org/jboss/marshalling/ByteOutput.
Warning: Could not resolve condition io.netty.handler.codec.memcache.AbstractMemcacheObjectAggregator for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.AbstractMemcacheObjectAggregator.
Warning: Could not resolve condition io.netty.handler.codec.memcache.AbstractMemcacheObjectDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.AbstractMemcacheObjectDecoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.AbstractMemcacheObjectEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.AbstractMemcacheObjectEncoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheDecoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec$Decoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec$Decoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec$Encoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheClientCodec$Encoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheObjectAggregator for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheObjectAggregator.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestDecoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheResponseDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheResponseDecoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheResponseEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheResponseEncoder.
Warning: Could not resolve condition io.netty.handler.codec.memcache.binary.BinaryMemcacheServerCodec for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.memcache.binary.BinaryMemcacheServerCodec.
Warning: Could not resolve condition io.netty.handler.codec.mqtt.MqttDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.mqtt.MqttDecoder.
Warning: Could not resolve condition io.netty.handler.codec.mqtt.MqttEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.mqtt.MqttEncoder.
Warning: Could not register io.netty.handler.codec.protobuf.ProtobufDecoder: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: com/google/protobuf/ExtensionRegistryLite.
Warning: Could not resolve condition io.netty.handler.codec.redis.RedisArrayAggregator for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.redis.RedisArrayAggregator.
Warning: Could not resolve condition io.netty.handler.codec.redis.RedisBulkStringAggregator for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.redis.RedisBulkStringAggregator.
Warning: Could not resolve condition io.netty.handler.codec.redis.RedisDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.redis.RedisDecoder.
Warning: Could not resolve condition io.netty.handler.codec.redis.RedisEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.redis.RedisEncoder.
Warning: Could not resolve condition io.netty.handler.codec.sctp.SctpInboundByteStreamHandler for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.sctp.SctpInboundByteStreamHandler.
Warning: Could not resolve condition io.netty.handler.codec.sctp.SctpMessageCompletionHandler for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.sctp.SctpMessageCompletionHandler.
Warning: Could not resolve condition io.netty.handler.codec.sctp.SctpMessageToMessageDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.sctp.SctpMessageToMessageDecoder.
Warning: Could not resolve condition io.netty.handler.codec.sctp.SctpOutboundByteStreamHandler for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.sctp.SctpOutboundByteStreamHandler.
Warning: Could not resolve condition io.netty.handler.codec.smtp.SmtpRequestEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.smtp.SmtpRequestEncoder.
Warning: Could not resolve condition io.netty.handler.codec.smtp.SmtpResponseDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.smtp.SmtpResponseDecoder.
Warning: Could not resolve condition io.netty.handler.codec.stomp.StompSubframeAggregator for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.stomp.StompSubframeAggregator.
Warning: Could not resolve condition io.netty.handler.codec.stomp.StompSubframeDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.stomp.StompSubframeDecoder.
Warning: Could not resolve condition io.netty.handler.codec.stomp.StompSubframeEncoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.stomp.StompSubframeEncoder.
Warning: Could not resolve condition io.netty.handler.codec.xml.XmlDecoder for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.codec.xml.XmlDecoder.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve condition io.netty.internal.tcnative.Library for reflection. Reason: java.lang.ClassNotFoundException: io.netty.internal.tcnative.Library.
Warning: Could not resolve org.apache.commons.logging.impl.Log4JLogger for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.Log4JLogger.
Warning: Could not resolve org.apache.commons.logging.impl.LogFactoryImpl for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl.
Warning: Could not resolve org.apache.commons.logging.impl.WeakHashtable for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.WeakHashtable.
Warning: Could not resolve org.apache.log4j.Level for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.log4j.Level.
Warning: Could not resolve org.apache.log4j.Priority for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.log4j.Priority.
Warning: Could not resolve org.apache.logging.log4j.core.LoggerContext for reflection configuration. Reason: java.lang.ClassNotFoundException: org.apache.logging.log4j.core.LoggerContext.
Warning: Could not resolve org.conscrypt.Conscrypt for reflection configuration. Reason: java.lang.ClassNotFoundException: org.conscrypt.Conscrypt.
Warning: Could not resolve org.conscrypt.OpenSSLContextImpl$TLSv13 for reflection configuration. Reason: java.lang.ClassNotFoundException: org.conscrypt.OpenSSLContextImpl$TLSv13.
Warning: Could not resolve org.hamcrest.number.OrderingComparison for reflection configuration. Reason: java.lang.ClassNotFoundException: org.hamcrest.number.OrderingComparison.
Warning: Could not resolve condition io.netty.handler.proxy.HttpProxyServer$HttpIntermediaryHandler for reflection. Reason: java.lang.ClassNotFoundException: io.netty.handler.proxy.HttpProxyServer$HttpIntermediaryHandler.
Warning: Could not register reactor.netty.channel.MicrometerChannelMetricsHandler$ConnectMetricsHandler: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: io/micrometer/core/instrument/Timer.
Warning: Could not register reactor.netty.channel.MicrometerChannelMetricsHandler$TlsMetricsHandler: queryAllPublicMethods for reflection. Reason: java.lang.NoClassDefFoundError: io/micrometer/core/instrument/Timer.
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing io.netty.channel.nio.AbstractNioChannel.<init>(io.netty.channel.Channel, java.nio.channels.SelectableChannel, int)
Parsing context:
   at io.netty.channel.nio.AbstractNioChannel.<init>(AbstractNioChannel.java:80)
   at io.netty.channel.nio.AbstractNioMessageChannel.<init>(AbstractNioMessageChannel.java:42)

        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:152)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:104)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:83)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraph(MethodTypeFlow.java:65)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultSpecialInvokeTypeFlow.onObservedUpdate(DefaultSpecialInvokeTypeFlow.java:61)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:558)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis$1.run(PointsToAnalysis.java:635)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of ch.qos.logback.classic.Logger are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=ch.qos.logback.classic.Logger.
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:140)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:257)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:228)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:105)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:84)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil$1.readValue(ConstantFoldUtil.java:55)
        at jdk.internal.vm.compiler/org.graalvm.compiler.core.common.spi.JavaConstantFieldProvider.readConstantField(JavaConstantFieldProvider.java:78)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantFieldProvider.readConstantField(AnalysisConstantFieldProvider.java:72)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil.tryConstantFold(ConstantFoldUtil.java:51)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.java.LoadFieldNode.asConstant(LoadFieldNode.java:178)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.java.LoadFieldNode.canonical(LoadFieldNode.java:144)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.java.LoadFieldNode.canonical(LoadFieldNode.java:135)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.java.LoadFieldNode.canonical(LoadFieldNode.java:72)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.spi.Canonicalizable$Unary.canonical(Canonicalizable.java:101)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.SimplifyingGraphDecoder.canonicalizeFixedNode(SimplifyingGraphDecoder.java:214)
        at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.canonicalizeFixedNode(PEGraphDecoder.java:1572)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.canonicalizeFixedNode(InlineBeforeAnalysis.java:192)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.SimplifyingGraphDecoder.handleFixedNode(SimplifyingGraphDecoder.java:193)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:821)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysis.java:240)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:548)
        at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:833)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:98)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:176)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:343)
        at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:93)
        ... 13 more
Error: Image build request failed with exit status 1
```

Thanks,
Wenqi

**Comment by snicoll** [comment]
These are all third-party hints and Netty still proves to be challenging. I am afraid this is isn't the right place to ask so I am going to close this now.

All I can say is that your Spring Cloud dependencies [do not match](https://start.spring.io/info) Spring Boot 3. You must use `2022.0.0-M4` or even the snapshot as the team is still working on native support there. For the native hints for Netty you're experimenting, it can be worth asking [on the reachability metadata repo](https://github.com/oracle/graalvm-reachability-metadata).

**Comment by sdeleuze** [comment]
Please make sure you are using GraalVM 22.2 since I don't see the error you mention when running `mvn -Pnative package`. On my side I see `Parameter 0 of constructor in com.github.wenqiglantz.service.demo.dataresidency.functions.LambdaFunctions required a bean of type 'reactivefeign.spring.config.ReactiveFeignNamedContextFactory' that could not be found.` on both JVM and native, potentially caused by the version of Spring Cloud.

Changing @snicoll sample to use `spring-boot-starter-webflux` works as expected.

**Comment by wenqiglantz** [comment]
Thanks so much for your input @sdeleuze & @snicoll!  I double checked my GraalVM version and it is indeed 22.2.  My spring cloud version was upgraded to 2022.0.0-M4 following @snicoll advice.  I was still running into the same error as listed in this issue.  I wondered if the root cause could be because of my windows OS as I ran my native image build on my windows laptop in x64 Native Tools Command Prompt (see attached the screenshot).  

To rule out Windows OS issue, I added a simple github actions workflow to run the native image build in my repo.  In this workflow, I have my runner set to run on ubuntu-latest.  I was glad to see the latest workflow run is indeed SUCCESSFUL!  It can be found here, https://github.com/wenqiglantz/spring-native-feign-reactive/actions/runs/3127280814/jobs/5073732809.  I have spring-cloud version updated to 2022.0.0-M4, GraalVM is on the latest version which is GraalVM CE 22.2.0 (build 17.0.4+8-jvmci-22.2-b06), which can be confirmed in the "print debug info" step in my workflow.  @sdeleuze, not sure why you were seeing the error you mentioned in the above comment.  Could it be OS related?  The inconsistent native image build behavior on different OS is a bit puzzling.  

I went ahead and deployed my native image in a custom runtime through a github actions workflow into AWS Lambda, upon testing, I got the following error (see cloudwatch screenshot attached for details): `Parameter 0 of constructor in com.github.wenqiglantz.service.demo.functions.LambdaFunctions required a bean of type 'com.github.wenqiglantz.service.demo.democlient.DemoClient' that could not be found`.  Any guidance on how to resolve this is greatly appreciated!

![image](https://user-images.githubusercontent.com/13189654/192309834-76ef335f-6d19-4b4e-b739-22d59a12504c.png)


Thanks again both for your help!!

Wenqi

![image](https://user-images.githubusercontent.com/13189654/192277380-0ca499a4-42ff-45a8-9cf2-39bb7239e877.png)
